src/main/kotlin/com/openlattice/linking/BackgroundLinkingService.kt

```diff
@@ -83,15 +81,17 @@ class BackgroundLinkingService(
                         )
                 )
 
-                val priority = priorityEntitySets.asSequence()
-
                 val rest = filteredLinkableEntitySetIds.filter {
                     !priorityEntitySets.contains(it)
                 }.asSequence()
 
+                val priority = priorityEntitySets.filter {
+                    filteredLinkableEntitySetIds.contains(it)
```

Comment:
Why not call asSequence first and apply the filter to the sequence?

Reply:
partially b/c the priorityEntitySets list is very small and fits in memory so there's little benefit to the laziness there. I can change it though

Reply:
For both priority and rest it will make a copy if you filter first. Not that big of deal.